src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/sentencing/SentencingAdjustmentsService.kt

```diff
@@ -24,27 +24,30 @@ class SentencingAdjustmentsService(
   retryQueueService = sentencingRetryQueueService,
 ) {
   suspend fun createAdjustment(createEvent: AdjustmentCreatedEvent) {
-    sentencingAdjustmentsMappingService.getMappingGivenAdjustmentId(createEvent.additionalInformation.id)
-      ?.let {
-        telemetryClient.trackEvent(
-          "sentencing-adjustment-create-duplicate",
-          mapOf(
-            "adjustmentId" to createEvent.additionalInformation.id,
-            "offenderNo" to createEvent.additionalInformation.nomsNumber,
-          ),
-          null,
-        )
-      } ?: let {
-      sentencingAdjustmentsApiService.getAdjustment(createEvent.additionalInformation.id).let { adjustment ->
-        if (adjustment.creatingSystem != CreatingSystem.NOMIS) {
+    synchronise {
+      name = "sentencing-adjustment"
+      telemetryClient = this@SentencingAdjustmentsService.telemetryClient
+      retryQueueService = this@SentencingAdjustmentsService.retryQueueService
+      eventTelemetry = mapOf(
+        "adjustmentId" to createEvent.additionalInformation.id,
+        "offenderNo" to createEvent.additionalInformation.nomsNumber,
+      )
+
+      checkMappingDoesNotExist {
+        sentencingAdjustmentsMappingService.getMappingGivenAdjustmentId(createEvent.additionalInformation.id)
+      }
+      transform {
+        val adjustment = sentencingAdjustmentsApiService.getAdjustment(createEvent.additionalInformation.id)
+
+        proceedWhen(adjustment.creatingSystem != CreatingSystem.NOMIS) {
```

Comment:
I'm not convinced about the need for this - is it possible for the entire transform function to return a nullable instead? null means we're not proceeding...

Also it might be easier to read if the entire transform parameter was a function call, e.g. `fun transformAdjustment(adjustmentId): SentenceAdjustmentMappingDto?`